python/graphstorm/model/utils.py

```diff
@@ -738,7 +756,106 @@ def save_shuffled_node_embeddings(shuffled_embs, save_embed_path, save_embed_for
 
     if rank == 0:
         with open(os.path.join(save_embed_path, "emb_info.json"), 'w', encoding='utf-8') as f:
-            f.write(json.dumps(emb_info))
+            json.dump(emb_info, f, indent=4)
+
+def save_full_node_embeddings(g, save_embed_path,
+                              embeddings,
+                              node_id_mapping_file,
+                              save_embed_format="pytorch"):
+    """ Save the entire node embeddings as pytorch tensors in
+        a distributed way.
+
+        Then input node `embeddings` are stored in Partition Node ID space.
+        By default, `save_full_node_embeddings` will translate the node IDs
+        from Partition Node ID space into their counterparts in Graph Node
+        ID space.
+
+        `save_full_node_embeddings` will save two information of an
+        embedding: 1) the embedding and 2) its corresponding node ID
+        in Graph Node ID space.
+        It assumes the input `embeddings` are stored in Partition Node
+        ID space and the IDs start from 0 to N. It will call NodeIDShuffler
+        to shuffle the node IDs from Partition Node ID space into Graph
+        Node ID space.
+
+        The saved node embeddings are in the following format:
+
+        Example
+        --------
+        # embedddings:
+        #   ntype0:
+        #     nids.part00000.bin
+        #     nids.part00001.bin
+        #     ...
+        #     emb.part00000.bin
+        #     emb.part00001.bin
+        #     ...
+        #   ntype1:
+        #     nids.part00000.bin
+        #     nids.part00001.bin
+        #     ...
+        #     emb.part00000.bin
+        #     emb.part00001.bin
+        #     ...
+
+        The content of nids.part files and emb.part files looks like:
+
+        Example:
```

Comment:
Can we separate the example into two. One for the nides.part file, and another for the emb.part file.